### ISSUE TITLE
`Breadcrumbs` should support `as` prop and custom label

### ISSUE THREAD

**Issue by iansan5653** [primer/react#3309]
[`Breadcrumbs`](https://primer.style/react/Breadcrumbs) always renders as `<nav aria-label="Breadcrumbs">...</nav>`. This is not ideal because `nav` creates a `navigation` landmark, and landmark roles should be used sparingly to avoid cluttering the screen reader navigation tools. And each instance of a landmark role should have a unique label, so there is currently no way to use multiple instances of the component on the same page.

As an example, you might have navigation menu that includes breadcrumbs along with a 'Home' link. In this case, you would want to include 'Home' inside the `navigation` landmark, so you might do something like:

```jsx
<nav aria-label="Site navigation">
  <a href="/index">Home</a>
  <Breadcrumbs>...</Breadcrumbs>
</nav>
```

This would result in semantically invalid HTML - a `nav` inside of a `nav`. In this case, it would be useful to render the breadcrumbs without the navigation role, like:

```jsx
<nav aria-label="Site navigation">
  <a href="/index">Home</a>
  <Breadcrumbs as="div" role="group">...</Breadcrumbs>
</nav>
```
But the `as` and `role` props are not allowed on this component.

Another example might be to use `Breadcrumbs` as a top-level site-wide navigation but also want another instance of the component to use for navigating within a source code view. In this case the second instance should not be a `nav` and/or should have a different label than "Breadcrumbs".

**Comment by lesliecdubs** [comment]
Hi @iansan5653 👋 Thanks for the report! Can you please share some more detail with us about the specific use case you have where you're running into a need for a `Home` anchor beside a `Breadcrumbs` component as opposed to inside of it?

A potential action item we might take here is to allow customizability for the `aria-label` value (e.g., don't hardcode `aria-label="Breadcrumbs"`), but would love to learn more about the use case first.
